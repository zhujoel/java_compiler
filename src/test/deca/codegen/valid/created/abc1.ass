; Test pour savoir si la pile est pleine
	TSTO #35
	BOV pile_pleine
	ADDSP #33
; start main program
; Class Declaration
; Code de la table des méthodes de Object
	LOAD #null, R0
	STORE R0, 20(GB)
	LOAD code.Object.equals, R0
	STORE R0, 21(GB)
; Main Function
; Declaration des variables
	LOAD #0, R0
	STORE R0, 22(GB)
	LOAD #0, R0
	STORE R0, 23(GB)
	LOAD #0, R0
	STORE R0, 24(GB)
	LOAD #0, R0
	STORE R0, 25(GB)
	LOAD #0, R0
	STORE R0, 26(GB)
	LOAD #0, R0
	STORE R0, 27(GB)
	LOAD #0x0.0p0, R0
	STORE R0, 28(GB)
	LOAD #0x0.0p0, R0
	STORE R0, 29(GB)
	LOAD #0x0.0p0, R0
	STORE R0, 30(GB)
	LOAD #0x0.0p0, R0
	STORE R0, 31(GB)
	LOAD #0x0.0p0, R0
	STORE R0, 32(GB)
; Beginning of main instructions:
	LOAD #54, R2
	STORE R2, 22(GB)
	LOAD #78987, R2
	STORE R2, 23(GB)
	LOAD #875, R2
	STORE R2, 24(GB)
	LOAD #99982, R2
	STORE R2, 26(GB)
; *
	LOAD 22(GB), R2
	PUSH R2
	LOAD #96, R2
	MUL R2, R2
	BOV debordement
	POP R2
	FLOAT R2, R2
	STORE R2, 28(GB)
; +
; *
	LOAD 28(GB), R2
	PUSH R2
	LOAD #7784, R2
	FLOAT R2, R2
	MUL R2, R2
	BOV debordement
	POP R2
	LOAD 26(GB), R2
	FLOAT R2, R2
	ADD R2, R2
	BOV debordement
	STORE R2, 29(GB)
; +
	LOAD 23(GB), R2
	PUSH R2
	LOAD 24(GB), R2
	ADD R2, R2
	BOV debordement
	POP R2
	FLOAT R2, R2
	STORE R2, 30(GB)
; +
; +
	LOAD 25(GB), R2
	PUSH R2
	LOAD 26(GB), R2
	ADD R2, R2
	BOV debordement
	POP R2
	FLOAT R2, R2
	LOAD 30(GB), R2
	ADD R2, R2
	BOV debordement
	STORE R2, 28(GB)
; +
; +
; +
	LOAD 22(GB), R2
	PUSH R2
	LOAD 23(GB), R2
	ADD R2, R2
	BOV debordement
	POP R2
	LOAD 24(GB), R2
	ADD R2, R2
	BOV debordement
	LOAD 26(GB), R2
	ADD R2, R2
	BOV debordement
	STORE R2, 27(GB)
; *
	LOAD 27(GB), R2
	PUSH R2
	LOAD 27(GB), R2
	MUL R2, R2
	BOV debordement
	POP R2
	FLOAT R2, R2
	STORE R2, 32(GB)
	HALT
code.Object.equals:
pile_pleine:
	WSTR "Erreur : Pile pleine."
	WNL
	ERROR
dereferencement_null:
	WSTR "Erreur : dereferencement null."
	WNL
	ERROR
tas_plein:
	WSTR "Erreur : Tas plein."
	WNL
	ERROR
debordement:
	WSTR "Erreur : debordement durant une operation arithmétique"
	WNL
	ERROR
io_error:
	WSTR "Erreur : input ou output"
	WNL
	ERROR
; end main program
