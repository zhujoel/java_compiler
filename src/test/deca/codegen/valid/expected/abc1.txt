Erreur : debordement durant une ope
